File: maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.go
import "fmt"
func maxScore(cardPoints []int, k int) int {
    n := len(cardPoints) - k
    left, right := 0, n - 1
    total := 0
    cur := 0
    for i, num := range cardPoints {
        total += num
        if i == n-1 {
            cur = total
        }
    }
    
    minimum := cur
    for right < len(cardPoints) - 1 {
        cur = cardPoints[right + 1] - cardPoints[left] + cur
        if cur < minimum {
            minimum = cur
        }
        left += 1
        right += 1
    }
​
    return total - minimum
}

File: maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int sum = 0;
        
        for (int i = 0; i < k; i++) {
            sum += cardPoints[i];
        }
        
        int max = sum;
        
        for (int i = 0; i < k; i++) {
            sum += cardPoints[cardPoints.length - 1 - i] - cardPoints[k - 1 - i];
            max = Math.max(sum, max);
        }
        
        return max;
    }
}